src/HDF5.jl

```diff
@@ -1412,10 +1414,16 @@ function Base.axes(cs::ChunkStorage{IndexCartesian})
     ntuple(i -> 1:chunk[i]:extent[i], length(extent))
 end
 
-function Base.setindex!(chunk_storage::ChunkStorage{IndexCartesian}, v::Tuple{<:Integer,Vector{UInt8}}, index::Integer...)
+# Filter flags provided
+function Base.setindex!(chunk_storage::ChunkStorage{IndexCartesian}, v::Tuple{<:Integer,AbstractArray}, index::Integer...)
```

Comment:
I think we probably should revisit the chunk interface and clean it up for v0.17.  For example, we can probably get rid of this or make it an internal/lower-level api. We can probably also handle the correct data layout conversion within the existing methods rather than forcing the user to handle it themselves.  Of course these would be breaking changes...

Reply:
I agree. We'll have to do some specification work to figure out what this looks like.

Perhaps the interface should not be specific to HDF5. We recently moved https://github.com/JuliaIO/Zarr.jl into JuliaIO.

I would be interested in an abstraction over both APIs and beyond since there is the potential to deal with raw and filtered chunks in both formats. @meggart proposed a common interface in https://discourse.julialang.org/t/common-interface-for-chunked-arrays/26009/10 and created https://github.com/meggart/ChunkedArrayBase.jl . This may be a good starting point.